### PR TITLE
CORCI-677 Functional test hardening

### PR DESCRIPTION
**Pull request by brianjmurrell**
More error and state checking.

Allow remote node account to be defined (i.e. in .localenv).

Patch avocado for tearDown timeout.
- https://github.com/avocado-framework/avocado/pull/3076

Allow ftest.sh to be used for just setup and teardown

### CODE REVIEW

File: ftest.sh [hunk 1]
```diff
@@ -64,9 +64,50 @@ restore_dist_files() {
 
 }
 
+cleanup() {
+    restore_dist_files "${yaml_files[@]}"
+    i=5
+    while [ $i -gt 0 ]; do
+        pdsh -l ${REMOTE_ACCT:-jenkins} -R ssh -S \
```
[the remark this answers is not in the document]
Comment (jolivier23) [reply ftest.sh]: is this just asking for "${REMOTE_ACCT:-jenkins}" ?   does that break it?

File: ftest.sh [hunk 2]
```diff
@@ -188,7 +217,8 @@ shift || true
 args+=" $*"
 
 # shellcheck disable=SC2029
-if ! ssh -i ci_key jenkins@"${nodes[0]}" "set -ex
+# shellcheck disable=SC2086
+if ! ssh $SSH_KEY_ARGS "${REMOTE_ACCT:-jenkins}"@"${nodes[0]}" "set -ex
```
[the remark this answers is not in the document]
Comment (jolivier23) [reply ftest.sh]: Why does this line have an odd number of quotes?   This is reaching the limits of my sh understanding.

File: ftest.sh [hunk 3]
```diff
@@ -217,6 +247,66 @@ dmesg
 df -h
 EOF
 
+# apply patch for https://github.com/avocado-framework/avocado/pull/3076/
```
Comment (jolivier23) [inline ftest.sh]: is this patch embedded in the shell script?   Why not just have a separate file?  Is it something we plan to push upstream?

Comment (brianjmurrell) [reply ftest.sh]: It's something that upstream has asked us to test to see if it resolves our issue, so yes, in time this patch should go away.

File: src/pool/srv_iv.c [hunk 4]
```diff
@@ -1,5 +1,5 @@
 /**
- * (C) Copyright 2017 Intel Corporation.
+ * (C) Copyright 2017Copyright 2017-2019 Intel Corporation.
```
Comment (jolivier23) [inline src/pool/srv_iv.c]: This doesn't look right but then I don't see you changed this file

Comment (brianjmurrell) [reply src/pool/srv_iv.c]: Bah.  Copyright update pre-commit hook seems to have run on the merge commit.  Which it clearly shouldn't.  I'll fix and push an update.

**Pull request merged**